Rework gui system

Grisshink opened the pull request:
The biggest problem that i'm coming across in the code is gui rendering. Every position of a button or a block has to be calculated manually and then drawn on the screen, which creates a lot of mess in the rendering part of codebase. So i'm working to rewrite this portion of code using my own gui library, which will suit all of my rendering needs and make the layout process very straightforward

planetashelezyaka commented:
ok, I'll try to review and check your changes when pull request will be finished. Currently I'm analyzing this code passively.

planetashelezyaka commented:
Removing nuklear gui code is very powerful step. Are you sure that it worth it? In this case many different ui components should be supported for this project such as: screens, text/digital boxes, combo boxes, color boxes, buttons and so on.
In my opinion Nuklear gui is not so huge and slow to replace it by own implementation. I think it would be better to leave gui approach unchanged in this project stage. When interface will be finished, you can make a decision about gui library.

Grisshink commented:
The biggest problem with Nuklear is its limited layout functionality. With Nuklear you can only draw with fixed size rows which does not allow porting the whole renderer to Nuklear (e.g. Blocks are impossible to do with Nuklear). Moreover, i have already implemented everything that the Nuklear does for my project, so i replaced it to unify the method of rendering the ui

planetashelezyaka commented:
Can we talk about this and others fixes? If it is ok for you, can you mail me?

lina-torovoltas reviewed (approved):
Everything is great, great code, I don’t see any problems and “Pbln”
